### DISCUSSION TITLE
How to build CefSharp with custom build cef

### DISCUSSION THREAD

**Discussion by SpikedPaladin** [cefsharp/CefSharp#4987] [Q&A] [answered]
I build cef with that commands
```set PATH=C:\code\depot_tools;%PATH%

set GN_DEFINES=symbol_level=1 is_official_build=true use_thin_lto=false proprietary_codecs=true ffmpeg_branding=Chrome enable_widevine=true
set GN_ARGUMENTS=--ide=vs2022 --sln=cef --filters=//cef/*

python3 automate-git.py --download-dir=c:\code\chromium_git --depot-tools-dir=c:\code\depot_tools --x64-build --with-pgo-profiles
```

Then i build cef-binary for x64 only using vs2022 and then added NuGet folder in it to VS nuget source.  
After that i opened CefSharp in VS used build.ps1 and after that i used [UpdateNugetPackages.ps1](https://github.com/cefsharp/CefSharp/blob/master/UpdateNugetPackages.ps1). It is finished without any errors. So i used build.ps1 once again and received that error.

```"C:\code\CefSharp-master\CefSharp3.sln" (rebuild target) (1) ->
"C:\code\CefSharp-master\CefSharp.Example\CefSharp.Example.csproj.metaproj" (Rebuild target) (2) ->
"C:\code\CefSharp-master\CefSharp.BrowserSubprocess\CefSharp.BrowserSubprocess.csproj.metaproj" (Rebuild target) (3) ->
"C:\code\CefSharp-master\CefSharp.BrowserSubprocess.Core\CefSharp.BrowserSubprocess.Core.vcxproj" (Rebuild target) (4) ->
(Link target) ->
  LINK : fatal error LNK1181: cannot open input file 'libcef.lib' [C:\code\CefSharp-master\CefSharp.BrowserSubprocess.Core\CefSharp.BrowserSubprocess.Core.vcxproj]
 
 
"C:\code\CefSharp-master\CefSharp3.sln" (rebuild target) (1) ->
"C:\code\CefSharp-master\CefSharp.Wpf.Example\CefSharp.Wpf.Example.csproj.metaproj" (Rebuild target) (6) ->
"C:\code\CefSharp-master\CefSharp.Wpf\CefSharp.Wpf.csproj.metaproj" (Rebuild target) (7) ->
"C:\code\CefSharp-master\CefSharp.Core.Runtime\CefSharp.Core.Runtime.vcxproj" (Rebuild target) (8) ->
  LINK : fatal error LNK1181: cannot open input file 'libcef.lib' [C:\code\CefSharp-master\CefSharp.Core.Runtime\CefSharp.Core.Runtime.vcxproj]
```

I found strange that it tries to lookup for libcef.lib but not libcef.dll.

Any fixes for that?

**Comment by amaitland** [comment]
When building `CEF`, use the instructions from https://bitbucket.org/chromiumembedded/cef/wiki/AutomatedBuildSetup.md#markdown-header-windows-configuration as a base. Looks like some of your build args are for a development build, not a release one.

I don't know what arguments you used for any of the powershell scripts, so it's not really possibly to provide any guidance past that.

> I found strange that it tries to lookup for libcef.lib but not libcef.dll.

It's a mixed mode `CLI/C++` library, linking a `.lib` file is expected.

**Reply by SpikedPaladin** [reply]
I used cef-binary script arguments from wiki and build error solved, at least for release build, but now when i updated nugets in WpfExample i can't run it because it exits with code 2147483651 (0x80000003). Also i tried to just replace all cef dlls in release folder, same result

**Reply by amaitland** [reply, marked as answer]
You should test with the `CEF Sample application (cefclient)` to confirm that your build actually works.
